effect/iter_performer.py

```diff
@@ -0,0 +1,49 @@
+from toolz import curry
+from functools import partial
+from effect import Effect, sync_performer
+from effect.do import do, do_return
+
+
+def iterator_performer(xs=None):
+    """
+    Returns performers for isolating iterator among other effects.
+    For exemplary usages one may look into tests.
+    """
+    xs = xs if xs is not None else []
+    @sync_performer
+    def performer(xs, dispatcher, intent):
+        """Chains yielded values and stores them in continuation"""
+        xs.append(intent)
+
+    @sync_performer
+    def retriever(xs, dispatcher, intent):
+        """Provides retrieval of gathered values"""
+        return xs
+
+    return partial(performer, xs), partial(retriever, xs)
```

Comment:
since `xs` is in the closure anyway, you don't need to explicitly take it as a parameter to `performer` and `retriever`, and don't need to explicitly `partial` it to bind it to those functions.